iohub/ngff/utils.py

```diff
@@ -0,0 +1,485 @@
+    - The function ensures that positions and channels are appended to an
+    existing plate if they are not already present.
+    """
+
+    # Limiting the chunking to 500MB
+    if chunks is None:
+        chunk_zyx_shape = _calculate_zyx_chunk_size(
+            shape,
+            np.dtype(dtype).itemsize,
+            max_chunk_size_bytes,
+        )
+
+        chunks = 2 * (1,) + chunk_zyx_shape
+
+    # Create plate
+    output_plate = open_ome_zarr(
+        str(store_path), layout="hcs", mode="a", channel_names=channel_names
+    )
+
+    # Create positions
+    for position_key in position_keys:
+        position_key_string = "/".join(position_key)
+        # Check if position is already in the store, if not create it
+        if position_key_string not in output_plate.zgroup:
+            position = output_plate.create_position(*position_key)
+            _ = position.create_zeros(
+                name="0",
+                shape=shape,
+                chunks=chunks,
+                dtype=dtype,
+                transform=[TransformationMeta(type="scale", scale=scale)],
+            )
+        else:
+            position = output_plate[position_key_string]
+
+    # Check if channel_names are already in the store, if not append them
+    for channel_name in channel_names:
+        metadata_channel_names = position.channel_names
+        if channel_name not in metadata_channel_names:
+            position.append_channel(channel_name, resize_arrays=True)
+
+
+def apply_transform_to_zyx_and_save(
+    func: Callable,
+    position_key: str,
+    input_store_path: Path,
+    output_store_path: Path,
+    input_channel_indices: Union[list[slice], list[list[int]]],
+    output_channel_indices: Union[list[slice], list[list[int]]],
+    input_time_indices: int,
+    output_time_indices: int,
+    **kwargs,
+) -> None:
+    """
+    Load a CZYX array from a Position object,
+    apply a transformation, and save the result.
+
+    Parameters
+    ----------
+    func : Callable
+        The function to be applied to the data.
+        Should take a CZYX array and return a transformed CZYX array.
+    position_key : str
+        The label of the position to process, e.g. "A/1/0".
+    input_store_path : Path
+        The path to input OME-Zarr Store.
+    output_store_path : Path
+        The path to output OME-Zarr Store.
+    input_channel_indices : Union[list[slice], list[list[int]]]
+        The channel indices to process. Acceptable values:
+        - A list of slices: [slice(0, 2), slice(2, 4), ...].
+        - A list of lists of integers: [[0], [1], [2, 3, 4], ...].
+        If empty list, process all channels.
+    output_channel_indices : Union[list[slice], list[list[int]]]
+        The channel indices to write to. Acceptable values:
+        - A list of slices: [slice(0, 2), slice(2, 4), ...].
+        - A list of lists of integers: [[0], [1], [2, 3, 4], ...].
```

Comment:
Is "[[0], [1], [2, 3, 4], ...]" a valid input? Can a single `func` take different numbers of channels? 

I *think* we need to specify that `func` takes a CZYX and returns a (C')ZYX, where C is the length of the inner `input_channel_indices` lists, and C' is the length of the `output_channel_indices`. Is my understanding correct?

Reply:
After reading closer, I'm a bit lost here. 

`input/output_channel_indices` can take non-nested `list[int]`, too, right?

It seems like the list of lists (or list of slices) types are only here to fix a slurmkit bug. If I understand correctly, I suggest we stick to the simpler `Union[list[int], slice]` type, which will be easier to understand. 

We can still handle the `slurmkit` bug as is.

Reply:
Fixed.